### PR TITLE
fix the rebootstrap issue

### PR DESCRIPTION
**Pull request by elgnay**

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go [hunk 1]
```diff
@@ -153,33 +171,64 @@ func (k *bootstrapController) sync(ctx context.Context, controllerContext factor
...
+			return nil
+		}
+	}
+
+	// otherwise add the rebootstrap annotation
+	klusterlet = klusterlet.DeepCopy()
+	if klusterlet.Annotations == nil {
+		klusterlet.Annotations = map[string]string{}
+	}
+	klusterlet.Annotations[helpers.RebootstrappingAnnotation] = "true"
```
Comment (qiujian16) [inline pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: why not set a rebootstrapProgressing condition?

Comment (elgnay) [reply pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: You are right. A condition looks better.

Comment (elgnay) [reply pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: Refactored with condition instead of annotation.

**Review by qiujian16 — COMMENTED** [review]
/approve

File: pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go [hunk 2]
```diff
@@ -153,33 +169,70 @@ func (k *bootstrapController) sync(ctx context.Context, controllerContext factor
...
+	)
+	if err != nil {
 		return err
 	}
-	ctrlContext.Recorder().Eventf("KlusterletAgentDeploymentDeleted", fmt.Sprintf("the deployment %s/%s is deleted due to %s",
-		namespace, workName, reason))
+	recorder.Eventf("KlusterletRebootstrap", fmt.Sprintf("The klusterlet %q starts rebootstrapping due to %s",
+		klusterletName, message))
+
+	// requeue and check the rebootstrap progress in 5 minutes
```
Comment (skeeey) [inline pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: 5 minutes or 5 seconds?

Comment (elgnay) [reply pkg/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: It should be `5 seconds`. Comment updated.

**Pull request merged**